pkg/pubsub/node_event_tracker.go

```diff
@@ -155,10 +168,18 @@ func (net *NodeEventTracker) Connected(n network.Network, c network.Conn) {
 			}
 		}
 	}
+
```

Comment:
This is surprising. Is there a situation where we have a panic to recover from? I'm interested about this block.

Reply:
was about to ask the same - when I see recover in the code just raise red flags here that we panic or don't handle pointers correctly somewhere.

Reply:
Now that I think about it, I'd already approved it, but needing `recover` here is a code smell. It might be warranted, but then I would appreciate an explanatory comment here.

Reply:
To be fair I'd avoid adding `recover` in our code because hides errors which are not correctly handled, but if you agree with it I'm fine. I'd second @mcamou here to leave a comment in the code at least on the reasoning if we are going to merge it.

Reply:
right, indeed my point is that by moving panics to logs makes them harder to surface until you analyze closely the logs, and run many instances of the binary. 

Making the application stop and yell the error out makes sure we catch these.